src/Rendering/prePassRenderer.ts

```diff
@@ -178,47 +216,32 @@ export class PrePassRenderer {
         this._engine = scene.getEngine();
 
         PrePassRenderer._SceneComponentInitialization(this._scene);
-        this._resetLayout();
+        this.defaultRT = this._createRenderTarget("sceneprePassRT", null);
+        this._setRenderTarget(null);
     }
 
-    private _initializeAttachments() {
-        const multiRenderLayout = [];
-        const clearLayout = [false];
-        const defaultLayout = [true];
-
-        for (let i = 0; i < this.mrtCount; i++) {
-            multiRenderLayout.push(true);
-
-            if (i > 0) {
-                clearLayout.push(true);
-                defaultLayout.push(false);
-            }
-        }
-
-        this._multiRenderAttachments = this._engine.buildTextureLayout(multiRenderLayout);
-        this._defaultAttachments = this._engine.buildTextureLayout(defaultLayout);
-    }
+    /**
+     * Creates a new PrePassRenderTarget
+     * This should be the only way to instanciate a `PrePassRenderTarget`
+     * @param name Name of the `PrePassRenderTarget`
+     * @param renderTargetTexture RenderTarget the `PrePassRenderTarget` will be attached to.
+     * Can be `null` if the created `PrePassRenderTarget` is attached to the scene (default framebuffer).
+     * @hidden
+     */
+    public _createRenderTarget(name: string, renderTargetTexture: Nullable<RenderTargetTexture>) : PrePassRenderTarget {
+        const rt = new PrePassRenderTarget(name, renderTargetTexture, { width: this._engine.getRenderWidth(), height: this._engine.getRenderHeight() }, 0, this._scene,
+            { generateMipMaps: false, generateStencilBuffer: true, defaultType: Constants.TEXTURETYPE_UNSIGNED_INT, types: [], drawOnlyOnFirstAttachmentByDefault: true });

```

Comment:
To save some resources, you can do: `generateStencilBuffer: true` => `generateStencilBuffer: this._engine.isStencilEnable` 

But I wonder if we really need to enable stencil at all on the `PrePassRenderTarget`?

Reply:
It is necessary for effects like `HighlightLayer`, but your remark about the engine level check is definitely valid